File: bindings/java/com/microsoft/cognitiveservices/speech/recognition/intent/IntentRecognitionResultEventArgs.java
package com.microsoft.cognitiveservices.speech.recognition.intent;
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//

/// <summary>
/// Define payload of intent intermediate/final result events.
/// </summary>
public class IntentRecognitionResultEventArgs // : System.EventArgs
{
    IntentRecognitionResultEventArgs(com.microsoft.cognitiveservices.speech.internal.IntentRecognitionEventArgs e)
    {
        this._Result = new IntentRecognitionResult(e.getResult());
        this._SessionId = e.getSessionId();
    }

    /// <summary>
    /// Represents the intent recognition result.
    /// </summary>
    public final IntentRecognitionResult getResult()
    {
        return _Result;
    }//{ get; }
    private IntentRecognitionResult  _Result;

    /// <summary>
    /// A String represents the session identifier.
    /// </summary>
    public final String getSessionId()
    {
        return _SessionId;
    }
    // { get; }
    private String _SessionId;

    /// <summary>
    /// Returns a String that represents the session id and the intent recognition result event.
    /// </summary>
    /// <returns>A String that represents the intent recognition result event.</returns>
    @Override
    public String toString()
    {
        return "SessionId:" + _SessionId + " ResultId:" + _Result.getResultId() + " Status:" + _Result.getReason() + " IntentId:<" + _Result.getIntentId() + "> Recognized text:<" + _Result.getRecognizedText() + ">."; 
    }
}

File: source/bindings/js/src/sdk/IntentRecognitionResultEventArgs.ts
//
// copyright (c) Microsoft. All rights reserved.
// licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//
import { IntentRecognitionResult, ISpeechProperties } from "./Exports";

/**
 * Intent recognition result event arguments.
 * @class
 */
export class IntentRecognitionResultEventArgs {
    /**
     * Represents the intent recognition result.
     * @property
     * @returns Represents the intent recognition result.
     */
    public result: IntentRecognitionResult;

    /**
     * A String represents the session identifier.
     * @property
     * @returns A String represents the session identifier.
     */
    public sessionId: string;

    /**
     * A set of properties.
     * @property
     * @returns A set of properties assoicated with this event.
     */
    public properties: ISpeechProperties;
}
